lib/pbench/agent/utils.py

```diff
@@ -390,3 +393,60 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+if os.environ.get("_PBENCH_UNIT_TESTS"):
+
+    class LocalRemoteHost:
+        """Mock out behavior for legacy unit tests.
+        """
+
+        def __init__(self):
+            self._local_names = frozenset(
+                [
+                    "127.0.0.1",
+                    "localhost",
+                    "172.16.42.42",
+                    "testhost.example.com",
+                    "testhost",
+                ]
+            )
+
+        def is_remote(self, host_name):
+            return host_name not in self._local_names
+
+
+else:
```

Comment:
Ugh. But, yeah; there's no way in the legacy test structure to mock external Python calls. ☹️ 

Even so, I'd prefer a smaller hack "mocking" `ifaddr.get_adapters` and `socket.getaddrinfo` so that we're testing most of the logic here.

Even if it's just something like

```
if os.environ.get("_PBENCH_UNIT_TESTS"):
    get_adapters = ifaddr.get_adapters
    getaddrinfo = socket.getaddrinfo
else:
    get_adapters = my_local_hack1
    getaddrinfo = lambda a, b: return a
```